src/test/java/org/openmrs/contrib/qaframework/automation/InpatienttSteps.java

```diff
@@ -0,0 +1,103 @@
+package org.openmrs.contrib.qaframework.automation;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.uitestframework.test.TestData;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+public class InpatienttSteps extends Steps {
+
+	private TestData.PatientInfo testPatient;
+
+	@Before(RunTest.HOOK.SELENIUM_ENCOUNTER)
+	public void visitHomePage() {
+		testPatient = createTestPatient();
+		initiateWithLogin();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_ENCOUNTER)
+	public void destroy() {
+		deletePatient(testPatient);
+		quit();
+	}
+
+	@Given("a user clicks on Find Patient app from Home page")
+	public void goToFindPatientRecordPage() {
+		findPatientPage = homePage.goToFindPatientRecord();
+	}
+
+	@Then("the system loads find patient record page")
+	public void systemloadsFindPatientRecordPage() {
+		assertTrue(textExists("Find Patient Record"));
+	}
+
+	@When("a user searches an existing patient from find patient record page")
+	public void searchExistingPatient() {
+		findPatientPage.enterPatient(testPatient.identifier);
+		findPatientPage.waitForPageToLoad();
+		dashboardPage = findPatientPage.clickOnFirstPatient();
+	}
+
+	@And("a user starts visit from the patient dashboard page")
+	public void clickOnStartVisit() {
+		visitsDashboardPage = dashboardPage.startVisit();
+	}
+
+	@Then("the system loads patient visit dashboard page")
+	public void systemloadsPatientVisitDashboardPage() {
+		assertTrue(visitsDashboardPage.containsText("Visits"));
+	}
+
+	// Admission encounter
+	@When("a user clicks on Admit to Inpatient button and selects location")
+	public void goToAdmitToInpatientPage() {
+		visitsDashboardPage.goToAdmitToInpatient().confirm("Inpatient Ward");
+	}
+
+	@And("the system confirms admission encounter is made")
+	public void confirmAdmissionEncounter() {
+		assertTrue(visitsDashboardPage.containsText("Admission"));
+		assertThat(visitsDashboardPage.getEncountersCount(), is(1));
+	}
+
+	// Transfer encounter
+	@When("a user clicks on Trasfer to Ward service button and selects location")
+	public void goToTrasferToWardServicePage() {
+		visitsDashboardPage.goToTransferToWardServicePage().confirm(
+				"Isolation Ward");
+	}
+
+	@Then("the system confirms trasfer within hospital encounter is made")
+	public void confirmTrasferEncounter() {
+		assertTrue(visitsDashboardPage.containsText("Transfer Within Hospital"));
+		assertThat(visitsDashboardPage.getEncountersCount(), is(2));
+	}
+
+	// Discharge encounter
+	@When("a user clicks on Exit from Inpatient button and selects location")
+	public void goToExitFromInpatientPage() {
+		visitsDashboardPage.goToExitFromInpatient().confirm("Isolation Ward");
```

Comment:
Can we use a different location here may be Inpatient Ward and get this merged.

Reply:
According to the test workflow, I mimicked a real world situation where a patient is admitted in [Inpatient-Ward](https://github.com/openmrs/openmrs-contrib-qaframework/blob/534d5e28cffa213f003ca12a625eff0795d47958/src/test/java/org/openmrs/contrib/qaframework/automation/InpatienttSteps.java#L63) then is transferred to [Isolation-Ward](https://github.com/openmrs/openmrs-contrib-qaframework/blob/534d5e28cffa213f003ca12a625eff0795d47958/src/test/java/org/openmrs/contrib/qaframework/automation/InpatienttSteps.java#L75) then is discharged from [Isolation-Ward](https://github.com/openmrs/openmrs-contrib-qaframework/blob/534d5e28cffa213f003ca12a625eff0795d47958/src/test/java/org/openmrs/contrib/qaframework/automation/InpatienttSteps.java#L88). By changing the location to Inpatient Ward at code line 88 would mean changing other correspoding Wards to bring out a real world user story and would proposed to go which the one suggested in the commit.